plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -3303,20 +3303,22 @@ void read_only::push_ro_transaction(const read_only::push_ro_transaction_params&
                } catch( chain::abi_exception& ) {
                   output = *trx_trace_ptr;
                }
-               const auto& accnt_metadata_obj = db.db().get<account_metadata_object,by_name>( params.account_name );
-               const auto& receipts = db.get_pending_trx_receipts();
-               vector<transaction_id_type>  pending_transactions;
-               pending_transactions.reserve(receipts.size());
-               for( transaction_receipt const& receipt : receipts ) {
-                  if( std::holds_alternative<transaction_id_type>(receipt.trx) ) {
-                     pending_transactions.push_back(std::get<transaction_id_type>(receipt.trx));
-                  }
-                  else {
-                     pending_transactions.push_back(std::get<packed_transaction>(receipt.trx).id());
+               if (db.is_building_block()){
+                  const auto& accnt_metadata_obj = db.db().get<account_metadata_object,by_name>( params.account_name );
+                  const auto& receipts = db.get_pending_trx_receipts();
+                  vector<transaction_id_type>  pending_transactions;
+                  pending_transactions.reserve(receipts.size());
+                  for( transaction_receipt const& receipt : receipts ) {
+                     if( std::holds_alternative<transaction_id_type>(receipt.trx) ) {
+                        pending_transactions.push_back(std::get<transaction_id_type>(receipt.trx));
+                     }
+                     else {
+                        pending_transactions.push_back(std::get<packed_transaction>(receipt.trx).id());
+                     }
                   }
+                  next(read_only::push_ro_transaction_results{db.head_block_num(), db.head_block_id(), db.last_irreversible_block_num(), db.last_irreversible_block_id(),
+                                                              accnt_metadata_obj.code_hash, pending_transactions, output});
```

Comment:
`next` always has to be called. It needs moved outside the `if`.

Reply:
Moved out of if, but compile failed. the parameter of next() need change, just paste the code that you want here to save the forth and back.